src/extension.ts

```diff
@@ -486,6 +486,207 @@ export function activate(context: vscode.ExtensionContext) {
+
+        const outputBuffers = [bufferData];
+        if (bufferOffset != bufferData.length) {
+            outputBuffers.push(new Buffer(bufferOffset - bufferData.length));
+        }
+
+        for (const accessorType of ['input', 'output']) {
+            const values = newData[accessorType];
+            const accessorComponents = accessorType == 'input' ? 1 : components;
+            const max = new Array(accessorComponents).fill(Number.NEGATIVE_INFINITY);
+            for (let i = 0; i < values.length; i++) {
+                const j = i % accessorComponents;
+                max[j] = Math.max(max[j], values[i]);
+            }
+            const min = new Array(accessorComponents).fill(Number.POSITIVE_INFINITY);
+            for (let i = 0; i < values.length; i++) {
+                const j = i % accessorComponents;
+                min[j] = Math.min(min[j], values[i]);
+            }
+            const float32Values = Float32Array.from(values);
+            const accessor = {
+                "bufferView": glTF.bufferViews.length,
+                "componentType": 5126,
+                "count": values.length / accessorComponents,
+                "type": accessorType == 'input' ? 'SCALAR' : samplerType,
+                "max": max,
+                "min": min
+            };
+            glTF.accessors[animationPointer.json[accessorType]] = accessor;
+            const newBufferView = {
+                "buffer": bufferView.buffer,
+                "byteOffset": bufferOffset,
+                "byteLength": float32Values.byteLength,
+            };
+            glTF.bufferViews.push(newBufferView);
+            if (alignedLength(float32Values.byteLength) != float32Values.byteLength) {
+                throw new Error('Float32Array not 4 byte length');
+            }
+            bufferOffset += float32Values.byteLength;
+            outputBuffers.push(Buffer.from(float32Values.buffer));
+        }
+
+        const createBufferFilename = !bufferJson.hasOwnProperty('uri') || bufferJson.uri.startsWith('data:');
+
+        let binFilename = bufferJson.uri;
+        if (createBufferFilename) {
+            let targetBasename = activeTextEditor.document.fileName;
+            if (path.extname(activeTextEditor.document.fileName).length > 1) {
+                const components = activeTextEditor.document.fileName.split('.');
+                components.pop();
+                targetBasename = components.join('.');
+            }
+            binFilename = targetBasename + '_data.bin';
+        }
+        bufferJson.uri = binFilename;
+
+        const dir = path.dirname(activeTextEditor.document.fileName);
+        const targetFilename = path.join(dir, binFilename);
+
+        const finalBuffer = Buffer.concat(outputBuffers);
+        fs.writeFileSync(targetFilename, finalBuffer, 'binary');
+
+        bufferJson.byteLength = finalBuffer.length;
+        const newJson = JSON.stringify(glTF, null, '  ');
```

Comment:
I wonder if we can respect the user's `editor.tabSize` and `editor.insertSpaces` settings here?  Something like this, converted from JS to TS...

```
var separator = useSpaces ? (new Array(tabSize + 1)).join(' ') : '\t';
```